packages/components/tests/lib/controllers/app-controller.test.js

```diff
@@ -1,839 +1,1389 @@
+  })
 
-    let testHomonym = {
-      lexemes: [
-        {
-          isPopulated: function () { return true },
-          languageID: LMF.getLanguageIdFromCode('lat')
-        }
-      ]
+  it('AppController - textSelectorParams: should return correct parameters if some other trigger is specified on desktop', async () => {
+    const triggerParams = {
+      a: 'value of A',
+      b: 'value of B'
     }
-    uiC.showLanguageInfo(testHomonym)
-    expect(uiC.panel.panelData.notification.visible).toBeTruthy()
-    expect(uiC.panel.panelData.notification.important).toBeFalsy()
-    expect(uiC.panel.panelData.notification.showLanguageSwitcher).toBeFalsy()
+    appC = AppController.jestCreate(uiState, { textQueryTriggerDesktop: triggerParams })
+    await appC.init()
+    appC._platform = { isMobile: false, isDektop: true }
+    const [event, eventParams] = appC.textSelectorParams
+    expect(event.name).toBe('GenericEvt')
+    expect(eventParams).toBe(triggerParams)
+  })
 
-    expect(uiC.popup.vi.popupData.notification.visible).toBeTruthy()
-    expect(uiC.popup.vi.popupData.notification.important).toBeFalsy()
-    expect(uiC.popup.vi.popupData.notification.showLanguageSwitcher).toBeFalsy()
+  it('AppController - registerTextSelector: should call registerSelector on the selection controller', async () => {
+    appC = AppController.jestCreate(uiState)
+    await appC.init()
+    const registerSelectorSpy = jest.spyOn(appC._selc, 'registerSelector')
+    appC.registerTextSelector()
+    expect(registerSelectorSpy).toBeCalledTimes(1)
+  })
 
-    // in this case, the language shown in the language notification should be
-    // the language actually tried, not the default from options
-    let testHomonymNoLexemes = {
-      languageID: LMF.getLanguageIdFromCode('grc')
-    }
-    uiC.showLanguageInfo(testHomonymNoLexemes)
-    expect(uiC.panel.panelData.notification.visible).toBeTruthy()
-    expect(uiC.panel.panelData.notification.important).toBeTruthy()
-    expect(uiC.panel.panelData.notification.showLanguageSwitcher).toBeTruthy()
-    expect(uiC.popup.vi.popupData.notification.text).toEqual(l10n.messages.TEXT_NOTICE_CHANGE_LANGUAGE.get(AppController.getLanguageName(testHomonymNoLexemes.languageID).name))
+  it('AppController - activateTextSelector: should call activateSelector on the selection controller', async () => {
+    appC = AppController.jestCreate(uiState)
+    await appC.init()
+    const activateSelectorSpy = jest.spyOn(appC._selc, 'activateSelector')
+    appC.activateTextSelector()
+    expect(activateSelectorSpy).toBeCalledTimes(1)
   })
 
-  it('7 AppController - showStatusInfo methods', () => {
-    uiC.showStatusInfo('fooSelectionText', latID)
-    expect(uiC.panel.panelData.status.languageName).toEqual('Latin')
-    expect(uiC.panel.panelData.status.selectedText).toEqual('fooSelectionText')
+  it('AppController - activateTextSelector: should call activateSelector on the selection controller', async () => {
```

Comment:
duplicate of prior test?

Reply:
`create()` should be called here without any parameter, will update